staging/staging.mk

```diff
@@ -0,0 +1,52 @@
+# staging.mk
+# this makefile is being included by staging/Makefile.include and staging/Makefile.dep.
+
+#
+# list of staging modules
+#
+# In order to mark a module as broken, add it to the "BROKEN" variable.
+# That way, it's dependees will be marked broken implicitly.
+#
+STAGING_MODULES += showcase
+#BROKEN += showcase    # (broken due to change in X, see #12345)
+
+#
+# list of staging applications
+#
+# In order to mark a staging application as "broken", turn it's
+# line into a comment (and add a note).
+#
+STAGING_APPS += showcase/hello-staging
+
+#
+# housekeeping
+#
+STAGING_MODULES := $(sort $(STAGING_MODULES))
+STAGING_MODULES_USED := $(filter $(USEMODULE),$(STAGING_MODULES))
+STAGING_APPS := $(sort $(STAGING_APPS))
+
+BROKEN := $(sort $(BROKEN))
+
+STAGING_MODULES_USED_BROKEN = \
+  $(sort $(filter $(BROKEN), $(STAGING_MODULES_USED)))
+
+STAGING_MODULES_USED_NOTBROKEN = \
+  $(sort $(filter-out $(BROKEN), $(STAGING_MODULES_USED)))
+
+# turn dependencies to broken staging modules into unsatisfied feature
+# requirements (broken-staging-module-<module_name>).
+# This will cause
+#  a) manually called "make" issue a warning
+#  b) "make info-boards-supported" skip any dependee, effectively disabling the CI
+#
+FEATURES_REQUIRED += $(patsubst %,broken-staging-module-%,$(STAGING_MODULES_USED_BROKEN))
```

Comment:
Still using `FEATURES_REQUIRED` as an error mechanism.

Reply:
Again: is that wrong because of the *naming* of variables, or because how the features mechanism actually works?

Reply:
FEATURES are not a way of providing an error message.

Reply:
It is used here mainly to provide "b) "make info-boards-supported" skip any dependee, effectively disabling the CI".

Reply:
That is what I mean, it is so that it is out of `make info-boards-supported` where this as nothing to do with it.

Reply:
So it is about the naming. If FEATURES would be called differently, e.g., BUILD_CHECKS, there's be no issue?

Do you have an alternative way of ensuring "b)"?

Reply:
Something like this yes, as I said in the first comment on this, it is a case of `invalid configuration`, and the `FEATURES` is only one part of it.

I am currently busy trying to have `make info-boards-supported` working correctly, and I do not plan on adding new functionality before it's done https://github.com/RIOT-OS/RIOT/issues/9913#issuecomment-434653782

Reply:
This is one make-line that does what's needed. How about keeping it for now and replace it with whatever facility you come up with later?

Reply:
@cladmi *ping*

Are you blocking this PR because of your ```make info-boards-supported``` rework?
Me and others are waiting for the staging directory.